src/huggingface_hub/hf_api.py

```diff
@@ -4849,6 +4833,7 @@ def upload_folder(
 
         ```
         """
+        repo_type = repo_type if repo_type is not None else REPO_TYPE_MODEL
```

Comment:
I would move this to `_validate_yaml` directly to prevent errors in the future in case we use it somewhere else

Reply:
Thanks, I've moved it to `_validate_yaml`.